UnityWeld/Binding/BindingHelper.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace UnityWeld.Binding
+{
+    public interface IComponentsCache : IDisposable{}
+
+    public interface IComponentsCache<T> : IComponentsCache
+    {
+        IList<T> Components { get; }
+    }
+
+    public static class BindingHelper
+    {
+        private class ComponentsCache<T> : IComponentsCache<T>
+        {
+            public IList<T> Components { get; }
+            public Type Type { get; }
+
+            public ComponentsCache(IList<T> cache)
+            {
+                Components = cache;
+                Type = typeof(T);
+            }
+
+            public void Dispose()
+            {
+                Components.Clear();
+                InternalCache[Type].Enqueue((IList)Components);
+            }
+        }
+
+        private static readonly Dictionary<Type, Queue<IList>> InternalCache = new Dictionary<Type, Queue<IList>>();
```

Comment:
Is this cache system necessary for the original purpose of making Unity Weld more decoupled from UI code? It looks like a separate feature to me but could maybe still be a useful addition. Could you explain a bit more about how it works?